auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -179,20 +218,23 @@ public Builder setSecret(String secret) {
 
         public IdpResponse build() {
             String providerId = mUser.getProviderId();
-            if ((providerId.equalsIgnoreCase(GoogleAuthProvider.PROVIDER_ID)
-                    || providerId.equalsIgnoreCase(FacebookAuthProvider.PROVIDER_ID)
-                    || providerId.equalsIgnoreCase(TwitterAuthProvider.PROVIDER_ID))
+            if (!AuthUI.SUPPORTED_PROVIDERS.contains(providerId)) {
+                throw new IllegalStateException("Unknown provider: " + providerId);
+            }
+            if ((providerId.equals(GoogleAuthProvider.PROVIDER_ID)
+                    || providerId.equals(FacebookAuthProvider.PROVIDER_ID)
```

Comment:
Maybe we should make `AuthUI.SOCIAL_PROVIDERS` or something so we can keep this list in one place in case it grows in the future?

Reply:
Good idea, done.